### PR TITLE
Add weighted bkava support for earn incentives

### PR DESCRIPTION
**Pull request by drklee3**

### CODE REVIEW

File: x/incentive/types/expected_keepers.go [hunk 1]
```diff
@@ -68,6 +68,15 @@ type SavingsKeeper interface {
...
+	IterateVaultRecords(ctx sdk.Context, cb func(record earntypes.VaultRecord) (stop bool))
+}
+
+// LiquidKeeper defines the required methods needed by this modules keeper
+type LiquidKeeper interface {
+	IsDerivativeDenom(ctx sdk.Context, denom string) bool
+	GetAllDerivativeDenoms(ctx sdk.Context) (denoms []string)
+	GetTotalDerivativeSupply(ctx sdk.Context) sdk.Int
+	GetDerivativeSupply(ctx sdk.Context, denom string) sdk.Int
 }
```
Comment (drklee3) [inline x/incentive/types/expected_keepers.go:80]: Some assumptions here on what methods the liquid keeper may have.

File: x/incentive/keeper/rewards_earn_accum_test.go [hunk 2]
```diff
@@ -86,11 +90,93 @@ func (suite *AccumulateEarnRewardsTests) TestStateUpdatedWhenBlockTimeHasIncreas
...
+	expectedIndexes := types.RewardIndexes{
+		{
+			CollateralType: "earn",
+			RewardFactor:   d("7.22"),
+		},
+		{
+			CollateralType: "ukava",
+			RewardFactor:   d("3.64"),
+		},
+	}
```
Comment (drklee3) [inline x/incentive/keeper/rewards_earn_accum_test.go:169]: This part is a bit confusing with multiple bkava vaults, different bkava vaults have the same per second per share saved in indexes assuming vault shares are 1:1 to bkava-xx supply.

For example with the following vaults with 100ukava per second set for "bkava"
* 95,000bkava-1
  * 95% of 100ukava per second == 95ukava per second
  * 95 / 95_000 == 0.001 per second per share
* 5,000bkava-2
  * 5% of 100ukava per second == 5ukava per second
  * 5 / 5_000 == 0.001 per second per share

After 1 hour:

  * rewards per share == 3600 seconds * 0.001ukava per second == 3.6ukava
  * bkava-1 vault
    * total rewards = 3.6ukava * 95,000 == 342000ukava
  * bkava-2 vault
    * total rewards = 3.6ukava * 5,000 == 18000ukava

  * total distributed to all bkava vaults == 360000ukava
  * 360000ukava / 3600seconds == 100ukava per second

Comment (rhuairahrighairidh) [reply x/incentive/keeper/rewards_earn_accum_test.go:169]: interesting

**Review by rhuairahrighairidh — APPROVED** [review]
I gave this one a failry surface level review and it looks good. I'd like to review it in more depth, but I think that'd be easier to do when reviewing the staking redistribution logic. So if you want to merge I'll add any issues I find to that PR.

**Pull request merged**